engine/src/test/java/org/kigalisim/engine/state/StreamKeeperTest.java

```diff
@@ -0,0 +1,494 @@
+ *
+ * @license BSD-3-Clause
+ */
+
+package org.kigalisim.engine.state;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.math.BigDecimal;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.kigalisim.engine.number.EngineNumber;
+import org.kigalisim.engine.number.StateGetter;
+import org.kigalisim.engine.number.UnitConverter;
+
+/**
+ * Tests for the StreamKeeper class.
+ */
+public class StreamKeeperTest {
+
+  /**
+   * Mock implementation of StateGetter for testing.
+   */
+  private static class MockStateGetter implements StateGetter {
+    private EngineNumber substanceConsumption = new EngineNumber(BigDecimal.ONE, "tCO2e / kg");
+    private EngineNumber energyIntensity = new EngineNumber(BigDecimal.ONE, "kwh / kg");
+    private EngineNumber amortizedUnitVolume = new EngineNumber(BigDecimal.ONE, "kg / unit");
+    private EngineNumber population = new EngineNumber(new BigDecimal("100"), "units");
+    private EngineNumber yearsElapsed = new EngineNumber(BigDecimal.ONE, "year");
+    private EngineNumber totalGhgConsumption = new EngineNumber(new BigDecimal("50"), "tCO2e");
+    private EngineNumber totalEnergyConsumption = new EngineNumber(new BigDecimal("100"), "kwh");
+    private EngineNumber volume = new EngineNumber(new BigDecimal("200"), "kg");
+    private EngineNumber amortizedUnitConsumption = 
+        new EngineNumber(new BigDecimal("0.5"), "tCO2e / unit");
+    private EngineNumber populationChange = new EngineNumber(new BigDecimal("10"), "units");
+
+    @Override
+    public EngineNumber getSubstanceConsumption() {
+      return substanceConsumption;
+    }
+
+    @Override
+    public EngineNumber getEnergyIntensity() {
+      return energyIntensity;
+    }
+
+    @Override
+    public EngineNumber getAmortizedUnitVolume() {
+      return amortizedUnitVolume;
+    }
+
+    @Override
+    public EngineNumber getPopulation() {
+      return population;
+    }
+
+    @Override
+    public EngineNumber getYearsElapsed() {
+      return yearsElapsed;
+    }
+
+    @Override
+    public EngineNumber getGhgConsumption() {
+      return totalGhgConsumption;
+    }
+
+    @Override
+    public EngineNumber getEnergyConsumption() {
+      return totalEnergyConsumption;
+    }
+
+    @Override
+    public EngineNumber getVolume() {
+      return volume;
+    }
+
+    @Override
+    public EngineNumber getAmortizedUnitConsumption() {
+      return amortizedUnitConsumption;
+    }
+
+    @Override
+    public EngineNumber getPopulationChange() {
+      return populationChange;
+    }
+  }
+
+  /**
+   * Mock implementation of OverridingConverterStateGetter for testing.
+   */
+  private static class MockOverridingConverterStateGetter 
```

Comment:
Please use Mockito instead. If this MockOverridingConverterStateGetter is no longer needed after bringing in Mockito, please remove it.